src/codeu/chat/client/commandline/Chat.java

```diff
@@ -151,8 +160,9 @@ public void invoke(Scanner args) {
     //
     panel.register("u-add", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
-        final String name = args.hasNext() ? args.nextLine().trim() : "";
+      public void invoke(List<String> args) {
+    	Iterator<String> stringIterator = args.iterator();
+        final String name = stringIterator.hasNext() ? stringIterator.next().trim() : "";
```

Comment:
Two comments:
1. Note that you're only taking the first token from the list. This means that if a user wants to, for example, send a message with more than one word, they have to enclose it in quotes. Maybe this is okay; it's up to you how you want to implement it! :)
2. In this case, an iterator is probably overkill. You can do something simpler like:
String name = args.isEmpty() ? "" : args.get(0).trim;
(Same comment applies to several other locations below too.)